backend/aws/lambda/app.py

```diff
@@ -76,31 +94,37 @@ def lambda_handler(event, context):
     if operation in operations:
         try:
             payload = load_payload(event)
-            response = respond(None, operations[operation](payload))
+            response = operations[operation](payload, conn)
         except Exception as e:
+            traceback.print_tb(e.__traceback__)
             print(e)
+            response = e
     else:
-        response = respond(ValueError('Unsupported method "{}"'.format(operation)))
+        response = Exception('Unsupported method "{}"'.format(operation))
+        payload = {}
+    
+    print(response)
 
     user_id = event['requestContext']['http']['sourceIp']
+    if isinstance(response, Exception):
+        base_id = str(uuid.uuid4())
+    else:
+        base_id = response['results'][0]['citeID']
+
     time_elapsed = time.time()-start
-    secret = get_secret()
 
     try:
-        secret = json.loads(secret)
-        conn = psycopg2.connect(host=secret['host'], user=secret['username'], password=secret['password'], database=secret['dbInstanceIdentifier'], port=secret['port'])
-
         cur = conn.cursor()
-        cur.execute("INSERT INTO deepcite_call (user_id,stage,status_code,response,response_time_elapsed,current_versions) VALUES (%s, %s, %s, %s, %s, %s)", (user_id,stage,response['statusCode'],response['body'],time_elapsed,json.dumps(versions)))
+        if new_submission:
+            cur.execute("INSERT INTO deepcite_retrieval (id,user_id,stage,status_code,response_time_elapsed,current_versions) VALUES (%s, %s, %s, %s, %s, %s)", (base_id, id, user_id,stage,response['statusCode'],time_elapsed,json.dumps(versions)))
+        else:
+            cur.execute("INSERT INTO deepcite_call (id,user_id,stage,status_code,response,response_time_elapsed,current_versions) VALUES (%s, %s, %s, %s, %s, %s, %s)", (base_id, id, user_id,stage,response['statusCode'],response['body'],time_elapsed,json.dumps(versions)))
         conn.commit()
     except psycopg2.OperationalError as e:
-        print("ERROR: Unexpected error: Could not connect to database instance.")
+        print("ERROR: Unexpected error: Could commit to database instance.")
         print(e)
-    else:
-        print("SUCCESS: Connection to RDS instance succeeded")
 
-    print(response)
-        
-    return response
+    return respond(payload.get('response_size', 'small'), response)
 
-# print(lambda_handler({"requestContext": {"http": {"method": "GET"}}},0))
+# body = "{\"id\": \"fdaeasfsd\", \"resonse_size\": \"small\", \"claim\":\"the death of Sherlock Holmes almost destroyed the magazine thries. When Arthur Conan Doyle killed him off in 1893, 20,000 people cancelled their subscriptions. The magazine barely survived. Its staff referred to Holmes’ death as “the dreadful event”.\", \"link\":\"http://www.bbc.com/culture/story/20160106-how-sherlock-holmes-changed-the-world\"}"
```

Comment:
I would be ok with having some of these as like `example.json` files especially if you think that would be useful for testing?